### PR TITLE
[WIP] Reduce bundled size by factoring away large libraries

### PR DESCRIPTION
**Pull request by killtheliterate**
Replaces:
- `underscore` with `lodash/whateverFunction`
- `request` with `axios`

### CODE REVIEW

**Comment by killtheliterate** [comment]
I don't know if this works yet, as tests don't seem to be running. Will `npm link` locally to see if it works in my particular case.

**Comment by killtheliterate** [comment]
So, this is working in my particular use-case. `axios` has a different api than `request`, so I've replaced the `request` with a `forEach` over `currentLogs`, which does an HTTP request for each log message and replaces the `currentLogs` with a modified array.

**Comment by billsaysthis** [comment]
Hi Garrett, thanks for the PR. I'm not familiar with Axios so I'm not sure why you switched it in. However one change that I don't think we can accept is is the `currentLogs.forEach()` loop since the point is to batch up log lines and only make one HTTP request per unit of time. Couple of smaller points but can you help me understand these two things first?

**Comment by killtheliterate** [comment]
@billsaysthis Heya, I swapped `axios` for `request` for the use-case where users are using a bundler, like Webpack. `request` produces very large builds, whereas `axios` results in a substantially smaller bundle. I think it's often the case nowadays that folks will use the module directly, instead of the browser specific version, and thought I'd propose this as a sort of rough-draft.

I certainly understanding the batching aspect, and if a smaller build (in the case of webapps) is something you're interested in, I could factor out the `forEach`... just kind of a rough-draft PR.

**Comment by billsaysthis** [comment]
I would definitely be open to just the `axios` change, though I would need some time to research and test to see if the switch can be made. The batching is an intentional feature, I don't see changing that since you can supply `0` as the sync interval to send one at a time.

**Comment by killtheliterate** [comment]
The `lodash` stuff is pretty much just drop in—it prevents the entire `underscore` lib from making it into a browser bundle. 

Another concern I have is not really knowing how `sumo` will deserialize what is sent (the `data` prop in the `axios` request). It'd be cool to have it be the same in both browser and node.

File: src/sumoLogger.js [hunk 1]
```diff
@@ -27,44 +29,52 @@ function setConfig(opts) {
...
     }
     if (currentConfig.hostName !== '') {
-      _.extend(headers, {'X-Sumo-Host': currentConfig.hostName});
+      assignIn(headers, {'X-Sumo-Host': currentConfig.hostName});
     }
 
-    request({
-      method: 'POST',
-      url: currentConfig.endpoint,
+    axios({
```
Comment (killtheliterate) [inline src/sumoLogger.js]: @billsaysthis This superficially looks similar to how `request` was being used, but I don't think `sumo` handles the body the same way. That's kind of a black box for me.

Relevant docs -> https://github.com/axios/axios#request-config

```javascript
// `data` is the data to be sent as the request body
// Only applicable for request methods 'PUT', 'POST', and 'PATCH'
// When no `transformRequest` is set, must be of one of the following types:
// - string, plain object, ArrayBuffer, ArrayBufferView, URLSearchParams
// - Browser only: FormData, File, Blob
// - Node only: Stream, Buffer
data: {
  firstName: 'Fred'
}
```

Comment (killtheliterate) [reply src/sumoLogger.js]: Looking through `request`, there's some type-ish checking with `Array.isArray()`

Comment (billsaysthis) [reply src/sumoLogger.js]: We have an internal hackathon in three weeks, I will block some time to verify this and https://github.com/SumoLogic/js-logging-sdk/pull/8 during that week.

**Comment by clementallen** [comment]
@billsaysthis was there any progress investigating this?  I am also using the module directly and relying on webpack to bundle it.  This is currently resulting in a gzipped size of over 300kb for the sumo-logger which is far too much.  Swapping to axios would be a massive help

**Comment by billsaysthis** [comment]
Implemented by https://github.com/SumoLogic/js-logging-sdk/pull/26 instead, thanks for this PR though.

**Pull request closed without merging**